extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -525,12 +533,20 @@ UnremovableBeanBuildItem unremovableRoutesBuilders() {
         }
 
         @Overridable
-        @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
+        @Record(value = ExecutionTime.STATIC_INIT, optional = true)
         @BuildStep(onlyIf = Flags.MainEnabled.class)
         CamelReactiveExecutorBuildItem reactiveExecutor(CamelMainRecorder recorder) {
```

Comment:
Like in initReactiveExecutor(), I'd appreciate a comment explaining why is this optional/overridable.